registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/MosipVerticleAPIManager.java

```diff
@@ -1,227 +1,239 @@
-		response.putHeader("content-type", contentType).putHeader("Access-Control-Allow-Origin", "*")
-				.putHeader("Access-Control-Allow-Methods", "GET, POST").setStatusCode(200)
-				.end(res);
-	}
-
-	// Added this method to cast all the stages to this class and invoke the deployVerticle method 
-	// to start the stage by configuration, since we don't want to test all the stages now, not marking this as
-	// an abstract method, but later this need to be marked as abstract
-	public void deployVerticle() {
-
-	}
-	
-
-	/**
-	 * Gets the stage name.
-	 *
-	 * @return the stage name
-	 */
-	protected String getStageName() {
-		return ClassUtils.getUserClass(this.getClass()).getSimpleName();
-	}
+package io.mosip.registration.processor.core.abstractverticle;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.util.ClassUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import brave.Tracing;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.virusscanner.spi.VirusScanner;
+import io.mosip.registration.processor.core.constant.HealthConstant;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.queue.factory.MosipQueue;
+import io.mosip.registration.processor.core.spi.queue.MosipQueueConnectionFactory;
+import io.mosip.registration.processor.core.spi.queue.MosipQueueManager;
+import io.mosip.registration.processor.core.tracing.VertxWebTracingLocal;
+import io.mosip.registration.processor.core.util.DigitalSignatureUtility;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.Json;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.micrometer.PrometheusScrapingHandler;
+
+/**
+ * @author Mukul Puspam
+ *
+ */
+public abstract class MosipVerticleAPIManager extends MosipVerticleManager {
+
+	@Value("${registration.processor.signature.isEnabled}")
+	Boolean isEnabled;
+
+	@Autowired
+	DigitalSignatureUtility digitalSignatureUtility;
+
+	@Autowired
+	ObjectMapper objectMapper;
+
+	@Autowired
+	private MosipQueueConnectionFactory<MosipQueue> mosipConnectionFactory;
+
+	@Autowired
+	private Tracing tracing;
+	
+	@Autowired
+	private Environment environment;
+
+	@Autowired(required = false)
+    private VirusScanner virusScanner;
+
+	/** The mosip queue manager. */
+	@Autowired
+	private MosipQueueManager<MosipQueue, byte[]> mosipQueueManager;
+
+	private static final String PROMETHEUS_ENDPOINT = "/actuator/prometheus";
+
+
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(MosipVerticleAPIManager.class);
+
+	@Value("${registration.processor.healthcheck.timeout}")

```

Comment:
Lets have the default time configured here itself as 5 seconds

Reply:
fixed